File: forest.R
# 2
library(readr)
# Importing the Dataset
forestfires <- read.csv('C:\\Users\\WIN10\\Desktop\\LEARNING\\DS\\Black Box (SVM)\\forestfires.csv', stringsAsFactors = TRUE)
summary(forestfires)
# Splitting the Data into train and test data
forestfires_train <- forestfires[1:256, ]
forestfires_test  <- forestfires[257:517, ]

# Training a model on the data 
# Begin by training a simple linear SVM
library(kernlab)
forestfires_classifier <- ksvm(size_category ~ ., data = forestfires_train, kernel = "vanilladot")

## Evaluating model performance ----
# predictions on testing dataset
forestfires_predictions <- predict(forestfires_classifier, forestfires_test)
table(forestfires_predictions, forestfires_test$size_category)
agreement <- forestfires_predictions == forestfires_test$size_category
table(agreement)
prop.table(table(agreement))

## Improving model performance ----
forestfires_classifier_rbf <- ksvm(size_category ~ ., data = forestfires_train, kernel = "rbfdot")
forestfires_predictions_rbf <- predict(forestfires_classifier_rbf, forestfires_test)
agreement_rbf <- forestfires_predictions_rbf == forestfires_test$size_category
table(agreement_rbf)
prop.table(table(agreement_rbf))

File: forest.py
# 2
import pandas as pd
import numpy as np
from sklearn.preprocessing import LabelEncoder
labelencoder = LabelEncoder()
v1 = pd.read_csv("C:\\Users\\WIN10\\Desktop\\LEARNING\\DS\\Black box\\forestfires.csv")
v1.describe()
v1.dtypes
v1['month']= labelencoder.fit_transform(v1['month'])
v1['day'] = labelencoder.fit_transform(v1['day'])
v1['size_category'] = labelencoder.fit_transform(v1['size_category'])
v1=pd.get_dummies(v1)
from sklearn.svm import SVC
from sklearn.model_selection import train_test_split
train,test = train_test_split(v1, test_size = 0.20)
train_X = train.iloc[:, 0:30]
train_y = train.iloc[:, -1]
test_X  = test.iloc[:, 0:30]
test_y  = test.iloc[:, -1]

# kernel = linear
help(SVC)
model_linear = SVC(kernel = "linear")
model_linear.fit(train_X, train_y)
pred_test_linear = model_linear.predict(test_X)
np.mean(pred_test_linear == test_y)

# kernel = rbf
model_rbf = SVC(kernel = "rbf")
model_rbf.fit(train_X, train_y)
pred_test_rbf = model_rbf.predict(test_X)
np.mean(pred_test_rbf==test_y)

# kernel = poly
model_linear = SVC(kernel = "poly")
model_linear.fit(train_X, train_y)
pred_test_linear = model_linear.predict(test_X)
np.mean(pred_test_linear == test_y)

# kernel = sigmoid
model_rbf = SVC(kernel = "sigmoid")
model_rbf.fit(train_X, train_y)
pred_test_rbf = model_rbf.predict(test_X)
np.mean(pred_test_rbf==test_y)
